app/models/question.rb

```diff
@@ -20,6 +34,26 @@ class Question < ActiveRecord::Base
 
   validates :text, presence: true
   validates :number, presence: true, numericality: { only_integer: true }
-
   validates :question_type, presence: true, inclusion: {in: %w[short_answer multiple_choice true_false]}
+  validates_with QuestionOrdersValidator, on: :next_question
+
+  def formatted_question_and_responses
+    # TODO (Chris) will implement later since it's pretty complex
+    # involves getting the responses too
+    text
+  end
+
+  # Assumes the response is valid
+  # Assumes the question orders is valid for this question valid?(:next_question) has been checked
+  def next_question(response)
+    if question_type == 'short_answer'
```

Comment:
We should avoid having multiple control flow paths depending on question_type; this will only get messier if number of question types grow. This is why we introduced the `DefaultQuestionOrder` and `ConditionalQuestionOrder` models. We should first check if a default question order exists.

Note that if a multiple choice question does not have diverging paths, it will not have any `ConditionalQuestionOrder` so `response_choice.question_order` will return nil. We can add attributes to the `ResponseChoice` model to grab the `DefaultQuestionOrder` through `question` but I don't think this is necessary if we do the check for default first as mentioned above.